lib/rspec/matchers/matcher_delegator.rb

```diff
@@ -1,8 +1,34 @@
 module RSpec
   module Matchers
+    # Provides a base class with as little methods as possible, so that
+    # most methods can be delegated via `method_missing`.
+    #
+    # On Ruby 2.0+ BasicObject could be used for this purpose, but it
+    # introduce some extra complexity with constant resolution, so the
+    # BlankSlate pattern was prefered.
+    # @private
+    class BaseDelegator
+      kept_methods = [
+        # Methods that raise warnings if removed.
+        :__id__, :__send__, :object_id,
+
+        # Methods that are explicitly undefined in some subclasses.
+        :==, :===,
+
+        # Methods we keep on purpose.
+        :class, :respond_to?, :__method__, :method, :dup,
+        :clone, :initialize_dup, :initialize_copy, :initialize_clone,
```

Comment:
Note that these are the methods required to pass the test suite, they don't all necessarily make sense to keep. e.g. `method` for instance is redundant with `__method__` and a cleanup of the test suite would allow to get rid of it.